fastavro/_reader.pyx

```diff
@@ -0,0 +1,628 @@
+# cython: auto_cpdef=True
+
+"""Python code for reading AVRO files"""
+
+# This code is a modified version of the code at
+# http://svn.apache.org/viewvc/avro/trunk/lang/py/src/avro/ which is under
+# Apache 2.0 license (http://www.apache.org/licenses/LICENSE-2.0)
+
+import json
+from struct import unpack, error as StructError
+from zlib import decompress
+import datetime
+from decimal import localcontext, Decimal
+from uuid import UUID
+
+from ._six import (
+    MemoryIO, xrange, btou, utob, iteritems, is_str, str2ints, fstint
+)
+from ._schema import (
+    extract_record_type, acquaint_schema, populate_schema_defs,
+    extract_logical_type
+)
+from .reader import SchemaResolutionError
```

Comment:
Can we change this so that it defines the its own `SchemaResolutionError`? I feel like if we are in `_foo.pyx` files then we should only import from other `.py` files if no `.pyx` exists. I'm not sure if there are any side effects from importing the regular python file, but it just feels a little worrisome.

Alternatively we could make a `_reader_common.py` that defines the `SchemaResolutionError` and both the py and pyx reader files import from there.

Reply:
Sounds reasonable. I will do it. I think technically things are probably okay as is, but at best it feels pretty odd. I had even considered whether it was possible to only install one set of files or the other. That's probably possible but pretty non-standard.

Reply:
I'm looking into this. In trying this, I'm running into a conflicting symbols issue again, where `fastavro.reader` is a module when the code wants a function. I think I had a solution for this earlier; I'll see if I can dig that up. Wacky interactions...

Reply:
I found a solution (or maybe just a workaround) for this. You'll see it in my next commit...

Reply:
Ironically, the workaround requires importing the regular `.py` files, not in order to use anything from them, but to prevent them from being imported later by other code in a problematic way. :-|